examples/basecoin/Makefile

```diff
@@ -1,41 +1,19 @@
 PACKAGES=$(shell go list ./... | grep -v '/vendor/')
 BUILD_FLAGS = -ldflags "-X github.com/cosmos/cosmos-sdk/examples/basecoin/version.GitCommit=`git rev-parse --short HEAD`"
 
-all: check_tools get_vendor_deps build test
-
-########################################
-### Build
+all: get_vendor_deps build test 
 
 build:
 	go build $(BUILD_FLAGS) -o build/basecoin ./cmd/...
 
-
```

Comment:
what happened here? Is this because we assume the tool stuff happens at the top level Makefile, so we can keep these smaller?

Reply:
I couldn't see how/why these tools were being used here. we should obviously not have duplicate tools either. (Can revert of course)

Reply:
The idea with the tools dir is that every root project have it, and all of our common Makefile and build tooling can go in there.

Really it should be a git submodule for each root project.  So, tendermint/tendermint has ./tools, and so does cosmos/cosmos-sdk/examples/basecoin, because both are root projects that require build tools.

So lets revert and use ./tools to handle things like... github.com/tendermint/glide installation, checking for proper golang version like ensuring it's at least version 1.9, etc... and later we can merge them into github.com/tendermint/make-tools and use it across our repos.